src/main/java/mat/server/Application.java

```diff
@@ -56,14 +57,14 @@
 import mat.server.util.MATPropertiesService;
 
 @Configuration
-@ComponentScan({"mat.model", "mat.dao", "mat.dao.impl", "mat.model.clause", "mat.server", "mat.hibernate"})
 @PropertySource("classpath:MAT.properties")
 @EnableTransactionManagement
 @EnableConfigurationProperties
 @EnableWebSecurity
 @EnableCaching
 @EnableScheduling
 @EnableJpaRepositories
+@Service
```

Comment:
why do you need a service annotation on Application class? It's probably not needed

Reply:
It's a war remember, not a spring boot application. That just ensures it gets scanned.

Reply:
I think it has "@Configuration" annotation which should be scanned by default.

Reply:
From [here](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html):

> @Configuration is meta-annotated with @Component, therefore @Configuration classes are candidates for component scanning